client/src/+assets/details.tsx

```diff
@@ -106,14 +60,48 @@ export const AssetDetails: NextPage = () => {
     return calculateStartEndPage(page, PROVENANCE_PER_PAGE)
   }, [page])
 
+  const getProvenanceInfo = async () => {
+    const published = await loadPublishedEvent(didZeroX(did), web3)
+    const bundlesPuchased = await getBundlesWithDataset(did)
+
+    const provenance = bundlesPuchased.map(bundle => {
+      return {
+        id: bundle.did,
+        action: 'bought',
+        date: new Date(bundle.createdAt),
+        account: bundle.user,
+        price: 0,
+        currency: 'USDC'
+      }
+    })
+
+    if (published) {
+      provenance.push({
```

Comment:
should published be first? in this case should be used `unshift` instead of `push`